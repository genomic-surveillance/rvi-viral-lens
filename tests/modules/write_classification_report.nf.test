nextflow_process {

    name "Test Process write_classification_report"
    script "modules/write_classification_report.nf"
    process "write_classification_report"

    test("Should run without failures") {

        when {
            process {
                """
                // define inputs of the process here. Example:
                input[0] = [
                    "test,2222222,SARSCov2,BA.2,,90,200,10,0.5\\n",
                    "test2,2222221,SARSCov2,BA.1,,80,500,20,0.02\\n"
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
