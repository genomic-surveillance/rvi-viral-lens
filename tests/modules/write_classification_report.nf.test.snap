{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "classification_report.csv:md5,cd03e5b2c6d19028d349a9caf9cc9b64"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-07-22T14:31:06.135939245"
    }
}